enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -2831,8 +2841,8 @@ Function *EnzymeLogic::CreatePrimalAndGradient(
 
     for (size_t i = 0; i < constant_args.size(); ++i) {
       auto arg = constant_args[i];
+      newArgs += 1;
```

Comment:
This should only be in the dup case, otherwise you increase twice for a constant arg (worthwhile adding a test for)